src/Core/Crawl/layout.cpp

```diff
@@ -0,0 +1,123 @@
+//#
+//# Copyright (C) 2021-2021 QuasarApp.
+//# Distributed under the GPLv3 software license, see the accompanying
+//# Everyone is permitted to copy and distribute verbatim copies
+//# of this license document, but changing it is not allowed.
+//#
+
+#include "layout.h"
+#include "clasteritem.h"
+#include <QtMath>
+
+namespace CRAWL {
+
+Layout::Layout() {
+
+}
+
+void Layout::add(ClasterItem *object) {
+
+    Claster::add(object);
+    updatePosition();
+
+}
+
+void Layout::remove(ClasterItem *object) {
+
+    Claster::remove(object);
+    updatePosition();
+
+}
+
+void Layout::changeLayout(const Refresh &fig) {
+    _shape = fig;
+    updatePosition();
+}
+
+void Layout::setDistance(int dist) {
+    _distance = dist;
+    updatePosition();
+}
+
+void Layout::updatePosition() {
+
+    switch (_shape) {
+        case CIRCLE:
+            drawCircle();
+            break;
+
+        case SQUARE:
+            drawSquare();
+            break;
+
+        case LINE:
+            drawLine();
+            break;
+
+        default:
+            break;
+     }
+
```

Comment:
you forget to invoke the refresh method after change the layout type 
```suggestion
     updatePosition();
```